### PR TITLE
Add Khronos official OpenCL ICD loader

### PR DESCRIPTION
**Pull request by uphoffc**

### CODE REVIEW

File: var/spack/repos/builtin/packages/opencl-icd-loader/package.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    depends_on("opencl-c-headers@2022.05.18", when="@2022.05.18")
+    depends_on("opencl-c-headers@2022.01.04", when="@2022.01.04")
+    depends_on("opencl-c-headers@2021.06.30", when="@2021.06.30")
+    depends_on("opencl-c-headers@2021.04.29", when="@2021.04.29")
+
+    provides("opencl@:3.0")
+
+    def cmake_args(self):
+        headers_include_dir = self.spec["opencl-c-headers"].prefix.include
+        args = [f"-DOPENCL_ICD_LOADER_HEADERS_DIR={headers_include_dir}"]
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/opencl-icd-loader/package.py]:
```suggestion
        args = [self.define("OPENCL_ICD_LOADER_HEADERS_DIR", headers_include_dir)]
```

Optional suggestion, but `self.define()` is a bit cleaner here.

**Review by alecbcs — DISMISSED** [review]
@uphoffc nice job on the PR! I added one optional suggestion, but otherwise this looks good to me. 

I swear we were working on a cleaner way to say that `PackageA` depends on the same version as `PackageB`, but I don't know if that's made it into develop yet. If so, that might help clean up your dependencies quite a bit.

**Comment by uphoffc** [comment]
@alecbcs Thanks for the review, I incorporated your suggestion.
Expressing that opencl-icd-loader always depends opencl-c-headers with the same version number would be great, but I could not find such a feature anywhere.

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks @uphoffc

**Pull request merged**